pkg/api/examples/static.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package examples
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+type staticExampleBuilder struct {
+	creater   runtime.ObjectCreater
+	convertor runtime.ObjectConvertor
+	examples  map[string]runtime.Object
+}
+
+var _ ExampleBuilder = &staticExampleBuilder{}
+
+func (f *staticExampleBuilder) NewExample(version, kind string) (runtime.Object, bool, error) {
+	if example, ok := f.examples[kind]; ok {
+		versionedExample, err := f.convertor.ConvertToVersion(example, version)
+		if err != nil {
+			return nil, true, err
+		}
+
+		return versionedExample, true, nil
+	}
+
+	example, err := f.creater.New(version, kind)
+	return example, false, err
+}
+
+func NewStaticExampleBuilder(creater runtime.ObjectCreater, convertor runtime.ObjectConvertor) ExampleBuilder {
+	return &staticExampleBuilder{
+		creater:   creater,
+		convertor: convertor,
+		examples: map[string]runtime.Object{
```

Comment:
Yeah.  So, in my mind, there's a couple of issues with that.

First, there's the immediate issue that the edit code (which is used by the `new` command) is structured around taking an object, serializing it, having the user edit, deserializing it, validating, and then reserializing for more edits.  Ergo, things like ordering would be lost after the first iteration of edits (or before, since the current code takes in the initial object, not a YAML string)

Second, there's the issue that just having a string of YAML could potentially allow an invalid object (either because of invalid YAML or because of invalid structure/values) to be checked into the repository, and then presented to the user.  The examples could/should probably be validated by unit tests, by why allow the extra potential for issues.

It's doable, but I'm not sure if it's the best solution currently.  Since we're concerned about comments, etc, I think it's probably a better idea in the long run to see if we can figure out how to get an annotating YAML serializer working -- this way, field descriptions get included automatically -- we don't have to write them by hand.